### PR TITLE
Add Iris Rev. 7

### PR DESCRIPTION
**Pull request by nooges**
## Description

Adds support for Iris Rev. 7

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebio/iris/rev7/config.h [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+#        define RGB_DISABLE_WHEN_USB_SUSPENDED
+#        define RGB_MATRIX_KEYPRESSES
+//#    endif
+#endif
+
+#define VIA_QMK_RGBLIGHT_ENABLE
+
+// Enable the workaround for the speed parameter mismatch between RGBLIGHT and
+// RGB Matrix, so that the speed slider in VIA behaves in a more useful way.
+#define VIA_CUSTOM_LIGHTING_ENABLE
```
Comment (drashna) [inline keyboards/keebio/iris/rev7/config.h:134]: Not needed, anymore.
```suggestion
```

File: keyboards/keebio/iris/rev7/rev7.c [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            if (data[1] == id_qmk_rgblight_effect_speed) {
+                rgb_matrix_set_speed_noeeprom(speed_from_rgblight(data[2]));
+            }
+            break;
+    }
+}
+
+#    endif // defined(VIA_ENABLE) && defined(VIA_CUSTOM_LIGHTING_ENABLE)
+
+#endif
```
Comment (drashna) [inline keyboards/keebio/iris/rev7/rev7.c:121]: This code is in core, and is automatically enabled for rgb matrix. 
```suggestion
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keebio/iris/rev7/config.h (hunk 1 above)
Comment (nooges) [reply keyboards/keebio/iris/rev7/config.h:134]: Sweet

**Review by drashna — APPROVED** [review]

File: keyboards/keebio/iris/rev7/config.h (hunk 1 above)
Comment (drashna) [reply keyboards/keebio/iris/rev7/config.h]: Yeah, I got tired of seeing the duplicated code appearing in a bunch of boards and basically slapped it into core.

File: keyboards/keebio/iris/keymaps/via/keymap.c [hunk 3]
```diff
@@ -65,3 +65,19 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
                                 // └────────┴────────┴────────┘                 └────────┴────────┴────────┘
   )
 };
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+#   if defined(KEYBOARD_keebio_iris_rev7)
```
Comment (nooges) [inline keyboards/keebio/iris/keymaps/via/keymap.c:71]: Looking for some feedback here on if this is the right approach for handling differing numbers of encoders

Comment (drashna) [reply keyboards/keebio/iris/keymaps/via/keymap.c:71]: Not sure there is a great way to handle this, aside from having the keymap folder in the different revision folders. 

But this should be okay, I think.

**Review by drashna — APPROVED** [review]

**Review by keyboard-magpie — APPROVED** [review]
Testing shows lots of historical user keymaps with RGBmatrix issues, main keymaps (default, via, miryoku) all behave normally.

**Pull request merged**